backend/sprintone/src/main/java/com/project/omega/bean/Product.java

```diff
@@ -0,0 +1,140 @@
+package com.project.omega.bean;
+
+import lombok.Builder;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.PositiveOrZero;
+import java.io.Serializable;
+
+@Entity
+@Builder
+public class Product implements Serializable {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long id;
+
+    @NotBlank
+    private String name;
+
+    @NotBlank
+    private String description;
+
+    @NotBlank
+    private String category;
+
+    @PositiveOrZero
+    private int price;
+
+    @OneToMany
+    @JoinColumn(name = "user_id", referencedColumnName = "id")
```

Comment:
I know I'm not a reviewer, but what is the reason for a product to be linked to a user? Surely it makes more sense to link it to a supplier?

Reply:
This was prior to the introduction of a supplier entity into our model, and is scheduled for change. The thinking here at the time was to track which user had added the product via introducing their id as a foreign key at the time of product creation, purely as we did not (at the time) have the notion of a supplier entity.